modules/tools/debugserver/Makefile

```diff
@@ -0,0 +1,59 @@
+###############################################################################
+#
+# Copyright 2016, Franco Bucafusco
+#
+# This file is part of CIAA Firmware.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+###############################################################################
+
+
+ifeq ($(ARCH),cortexM4)
+include modules$(DS)tools$(DS)debugserver$(DS)openocd$(DS)mak$(DS)Makefile
+else
+
+ifeq ($(ARCH),cortexM0)
+include modules$(DS)tools$(DS)debugserver$(DS)openocd$(DS)mak$(DS)Makefile
+else
+
+ifeq ($(ARCH),msp430)
+include modules$(DS)tools$(DS)debugserver$(DS)mspdebug$(DS)mak$(DS)Makefile
+else
+
+ifeq ($(ARCH),x86)
+#nothing
+else
+ifeq ($(ARCH),mips)
+#nothing
+else
+#error
+endif
+endif
+endif
+endif
+endif
```

Comment:
si, lo estoy cambiando.

On 07/09/2016 02:55 PM, Juan Cecconi wrote:
>
> In modules/tools/debugserver/Makefile 
> <https://github.com/ciaa/Firmware/pull/431#discussion_r70170128>:
>
> > +ifeq ($(ARCH),msp430)
> > +include modules$(DS)tools$(DS)debugserver$(DS)mspdebug$(DS)mak$(DS)Makefile
> > +else
> > +
> > +ifeq ($(ARCH),x86)
> > +#nothing
> > +else
> > +ifeq ($(ARCH),mips)
> > +#nothing
> > +else
> > +#error
> > +endif
> > +endif
> > +endif
> > +endif
> > +endif
>
> Para mi esto sigue siendo feo...
> Yo decía algo como esto
>
> https://github.com/ciaa/Firmware/blob/master/modules/tools/openocd/mak/Makefile
>
> Y cada mk incluye lo suyo
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/ciaa/Firmware/pull/431/files/693b4e863d0fc61cacb5cccd9405ffc3f36e01e8..615d43ab217be7022671da1fe278e3051920aed3#r70170128>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AQg2FlHfVxG6mpEEAzVzBXI_zLgSAZhAks5qT-B2gaJpZM4JFml7>.
>